contract/src/main/java/io/neow3j/contract/NeoNameService.java

```diff
@@ -0,0 +1,368 @@
+     * {@link TransactionBuilder} based on this script.
+     * <p>
+     * Only committee members are allowed to add a new root domain.
+     *
+     * @param root the new root domain.
+     * @return a transaction builder.
+     */
+    public TransactionBuilder addRoot(String root) {
+        checkRegexMatch(ROOT_REGEX_PATTERN, root);
+        return invokeFunction(ADD_ROOT, string(root));
+    }
+
+    /**
+     * Creates a transaction script to set the price for registering a domain and initializes a
+     * {@link TransactionBuilder} based on this script.
+     * <p>
+     * Only committee members are allowed to set the price.
+     *
+     * @param price the price for registering a domain.
+     * @return a transaction builder.
+     */
+    public TransactionBuilder setPrice(BigInteger price) {
+        if (!isValidPrice(price)) {
+            throw new IllegalArgumentException("The price needs to be greater than 0 and smaller " +
+                    "than 1_000_000_000_000.");
+        }
+        return invokeFunction(SET_PRICE, integer(price));
+    }
+
+    // true if the price is in the allowed range, false otherwise.
+    private boolean isValidPrice(BigInteger price) {
+        return price.compareTo(BigInteger.ZERO) > 0 &&
+                price.compareTo(MAXIMAL_PRICE) <= 0;
+    }
+
+    /**
+     * Gets the price to register a domain.
+     *
+     * @return the price to register a domain.
+     * @throws IOException if there was a problem fetching information from the Neo node.
+     */
+    public BigInteger getPrice() throws IOException {
+        return callFuncReturningInt(GET_PRICE);
+    }
+
+    /**
+     * Checks if the specified second domain name is available.
+     *
+     * @param name the domain name.
+     * @return true if the domain name is available, false otherwise.
+     * @throws IOException if there was a problem fetching information from the Neo node.
+     */
+    public boolean isAvailable(String name) throws IOException {
+        checkDomainNameValidity(name);
+        try {
+            return callFuncReturningBool(IS_AVAILABLE, string(name));
+        } catch (IndexOutOfBoundsException e) {
+            String root = name.split("\\.")[1];
+            throw new IllegalArgumentException("The root domain '" + root + "' does not exist.");
+        }
+    }
+
+    /**
+     * Creates a transaction script to register a new domain and initializes a
+     * {@link TransactionBuilder} based on this script.
+     *
+     * @param name  the domain name.
+     * @param owner the address of the domain owner.
+     * @return a transaction builder.
+     */
+    public TransactionBuilder register(String name, ScriptHash owner) throws IOException {
+        checkDomainNameAvailability(name, true);
+        return invokeFunction(REGISTER, string(name), hash160(owner));
+    }
+
+    // checks if a domain name is available
+    private void checkDomainNameAvailability(String name, boolean shouldBeAvailable)
```

Comment:
Maybe it'd be a good idea to change `shouldBeAvailable` to `isForRegistration`.

I think `isForRegistration` it's a bit more legible than `shouldBeAvailable` since `true` or `false` depends if it's for `renew` (or related actions) or for `register`.

Reply:
I agree, I'll change it to `isForRegistration`.